docs/guides/tools.md

```diff
@@ -45,19 +45,17 @@ type Query {
 
 ---
 
-### `depth_limit_validator`
+### `WithQueryDepthLimiter`
 
-Create a validator to limit the complexity of queries by their depth to protect against malicious
+Add a query depth limter validation rule that limits the complexity of queries by their depth to protect against malicious
 queries.
 
 ```python
-from graphql import ValidationRule
-
-def depth_limit_validator(
+def WithQueryDepthLimiter(
```

Comment:
I'm a bit unsure on why this is a function and not a class, not really a fan of using camel case names for functions 😊

Reply:
Yeah I agree, I was just trying to match the current naming convention. Happy to make them snake case. It would look something like this:

```python
schema = strawberry.Schema(
  Query,
  extensions=[
    with_query_depth_limiter(max_depth=4),
    MyCustomExtension,
    use_validation_cache(),
  ]
)
```

Alternative is that we implement a `.build()` class method?

```python
schema = strawberry.Schema(
  Query,
  extensions=[
    QueryDepthLimiter.build(max_depth=4),
    MyCustomExtension,
    ValidationCache.build(),
  ]
)
```

Which looks the best @patrick91 ?

Reply:
What prevents us from doing this?

```
schema = strawberry.Schema(
  Query,
  extensions=[
    QueryDepthLimiter(max_depth=4),
    MyCustomExtension, # or MyCustomExtension() (or we instanciate it if not already)
    ValidationCache(),
  ]
)
```

Reply:
It makes the code more convoluted because the extensions get instantiated later with the execution context. The depth limiter extension would look like this:

```python
class QueryDepthLimiter(Extension):
	def __init__(
		self, 
		max_depth: int,
    	ignore: Optional[List[IgnoreType]] = None,
    	callback: Callable[[Dict[str, int]], None] = None,
	):
		self.validation_rule = create_validator(max_depth, ignore, callback)

	def __call__(self, execution_context):
		self.execution_context = execution_context

	def on_request_start(self):
    	self.execution_context.validation_rules = (
        	self.execution_context.validation_rules + (self.validation_rule,)
        )
```

IMO creating a factory method is more intuitive for me but I'm also happy to change the extensions to this format.

Reply:
let's see what @BryceBeagle thinks 😊

Reply:
I've refactored the extensions to use classes.